### PR TITLE
Use committed certificate round instead of highest (last) committed round to remove certificates from DAG

### PR DESCRIPTION
**Pull request by mwtian**
After a sub-dag is selected, each of its certificate will be committed and removed from the DAG. The current cleanup algorithm also removes all certificates at origin `A` from rounds below `r` from the DAG, when committing a certificate from origin `A` at round `r`.

A mistake was made in https://github.com/MystenLabs/sui/commit/2eb8979218e4dac844c5ab539e93c6dd6aab286c. I tried to simplify the cleanup logic to only remove certificates from the committed certificate's own origin, instead of going through all origins and their last committed rounds. However I used the overall last committed round instead of the round of certificate for cleanup. This change fixes the mistake.

Further simplifications and testing of the logic will definitely be needed. But I think they belong to another PR.

### CODE REVIEW

File: narwhal/consensus/src/consensus.rs [hunk 1]
```diff
@@ -173,7 +173,7 @@ impl ConsensusState {
             .retain(|r, _| r + gc_depth >= self.last_committed_round);
         // Also purge all certificates at and below last committed round of the certificate origin
```
Comment (gdanezis) [inline narwhal/consensus/src/consensus.rs:174]: Maybe fix the comment too?

**Review by gdanezis — APPROVED** [review]
Nice one! One minor nit to fix the comment.

**Review by akichidis — APPROVED** [review]
Ah good catch @mwtian ! Can't believe how I missed this in the earlier review.

Looking closer to the failing simtest it seems that we did run in a situation where certificates where delivered to the respective node dags with different latency and practically now certificates started get dropped in a non deterministic way (some pruned earlier than others). 

100% agree on more testing

File: narwhal/consensus/src/consensus.rs (hunk 1 above)
Comment (mwtian) [reply narwhal/consensus/src/consensus.rs:174]: Updated this and other comments.

**Pull request merged**